pyartcd/pyartcd/pipelines/ocp4.py

```diff
@@ -1,13 +1,81 @@
+
 import traceback
 
 import click
 from aioredlock import LockError
 
-from pyartcd import locks
+from pyartcd import locks, util, plashets
 from pyartcd.cli import cli, pass_runtime, click_coroutine
 from pyartcd.runtime import Runtime
 from pyartcd.s3 import sync_repo_to_s3_mirror
 
+# Doozer working dir
+DOOZER_WORKING = 'doozer_working'
+
+
+@cli.command("ocp4:build-compose",
+             help="If any RPMs have changed, create multiple yum repos (one for each arch) based on -candidate tags"
+                  "those repos can be signed (release state) or unsigned (pre-release state)")
+@click.option('--version', required=True, help='Full OCP version, e.g. 4.14-202304181947.p?')
+@click.option('--assembly', required=True, help='The name of an assembly to rebase & build for')
+@click.option('--build-rpms', is_flag=True, help='True if RPMs should be built')
+@pass_runtime
+@click_coroutine
+async def build_compose(runtime: Runtime, version: str, assembly: str, build_rpms: bool):
+    stream_version, release_version = version.split('-')  # e.g. (4.14, 202304181947.p?) from 4.14-202304181947.p?
+
+    # Create a Lock manager instance
+    lock_policy = locks.LOCK_POLICY['compose']
+    lock_manager = locks.new_lock_manager(
+        internal_lock_timeout=lock_policy['lock_timeout'],
+        retry_count=lock_policy['retry_count'],
+        retry_delay_min=lock_policy['retry_delay_min']
+    )
+    lock_name = f'compose-lock-{stream_version}'
+
+    # Build compose
+    try:
+        async with await lock_manager.lock(lock_name):
+            automation_state: str = await util.get_freeze_automation(stream_version)
+            runtime.logger.info('Automation state for %s: %s', stream_version, automation_state)
+
+            # - If automation is not frozen, go ahead
+            # - If automation is "scheduled" and job was triggered by human
+            #   and there were no RPMs in the build plan: Do not build compose
+            # - If automation is "scheduled" and job was triggered by human
+            #   and there were RPMs in the build plan: Build compose, and warn
+            if automation_state in ['scheduled', 'yes', 'True'] or \
+                    (automation_state in ['scheduled', 'weekdays'] and build_rpms):
+                mirror_plashet = await plashets.build_plashets(
+                    stream_version, release_version, assembly=assembly, dry_run=runtime.dry_run)
+                click.echo(mirror_plashet)
+            else:
+                runtime.logger.info('Skipping compose build')
+
+            if automation_state == 'scheduled' and build_rpms:
```

Comment:
Wondering why this notification block is separate than the one above? Would like to merge these two to keep it simple to understand

Reply:
Also done in 195e55f5e4b764822c72aae4a88e16026a3fc568